assets/scripts/dialogs/__tests__/GeotagDialog.test.js

```diff
@@ -54,43 +70,66 @@ describe('GeotagDialog', () => {
     expect(wrapper.queryByText('Confirm location')).not.toBeInTheDocument()
   })
 
-  // TODO: Fix these tests
-  it.skip('allows a location to be confirmed when the current signed-in user is the street owner', () => {
+  /*
+  the tests for location pop up already test whether the popup shows up or not
+  tests for StreetMetaGeotag also cover some of this behavior
+  do we have a setup available for mocking a signed-in vs anonymous user? */
+  it('allows a location to be cleared when the current signed-in user is the street owner', () => {
     isOwnedByCurrentUser.mockReturnValueOnce(true)
-    const wrapper = renderWithReduxAndIntl(<GeotagDialog />, { initialState })
+    const wrapper = renderWithReduxAndIntl(<GeotagDialog />, {
+      initialState
+    })
 
-    expect(wrapper.queryByText('Confirm location')).toBeInTheDocument()
+    expect(wrapper.queryByText('Clear location')).toBeInTheDocument()
   })
 
-  it.skip('allows a location to be confirmed when the current anonymous user started this street', () => {
+  /* this test is literally the same as the above test...so again its either redundant or missing something?
+  we can also test more things here, like actually confirming/adding the new location, though thats also
+  covered a little bit by cypress at this point
+  */
+  it.skip('allows a location to be cleared when the current anonymous user started this street', () => {
     isOwnedByCurrentUser.mockReturnValueOnce(true)
     const wrapper = renderWithReduxAndIntl(<GeotagDialog />, { initialState })
 
+    expect(wrapper.queryByText('Clear location')).toBeInTheDocument()
+  })
+
+  it('allows a location to be added when the current anonymous user started this street', () => {
+    isOwnedByCurrentUser.mockReturnValueOnce(true)
+    const wrapper = renderWithReduxAndIntl(<GeotagDialog />, {
+      initialState: {
+        ...initialState,
+        street: {
+          creatorId: 'mayorofnullisland',
+          location: null
+        }
+      }
+    })
+
     expect(wrapper.queryByText('Confirm location')).toBeInTheDocument()
   })
 
-  it('does not allow a location to be confirmed when the current signed-in user is not the street owner', () => {
+  /* neither confirm or clear location buttons should show up in this case */
+  it('does not allow a location to be cleared when the current user is not the street owner', () => {
     isOwnedByCurrentUser.mockReturnValueOnce(false)
     const wrapper = renderWithReduxAndIntl(<GeotagDialog />, { initialState })
-
-    expect(wrapper.queryByText('Confirm location')).not.toBeInTheDocument()
+    expect(wrapper.queryByTestId('locationEdit')).not.toBeInTheDocument()
```

Comment:
Neither 'Confirm Location' or 'Clear Location' show up in this case, so there is no text to search for. My instinct that case was basically to instead just test whether that block was in the document. 

I could re-write it to use queryByText and check that _both_ 'Confirm location' and 'Clear location' don't exist, but this felt more straightforward and also makes the component behavior a bit more legible through the test, IMO. For instance, I didn't really even realize that this was the case despite having refactored the whole thing, but now someone else or future me could look at this test and see that we don't expect to render those buttons in this particular scenario. I'm open to other options, though.

Reply:
I'd go with what RTL would recommend here. The [`queryBy*` functions, unlike `getBy*`, don't throw errors if something isn't found, so they're designed to do what we're doing with it](https://testing-library.com/docs/dom-testing-library/api-queries#queryby):

> This is useful for asserting an element that is not present.

And they do [recommend _not_ using a `testid` when other queries are appropriate](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-querying-by-text). (Thanks for sending that link!)

> I recommend you query by the actual text (in the case of localization, I recommend the default locale), rather than using test IDs or other mechanisms everywhere. 

I expect the "official" recommendation to take precedence in this decision because these are the resources someone is likely to be looking to for guidance, or have learned RTL from -- we would be continuously trying to defend the `testid` usage against the recommendation. (Let me know if you still disagree with it, though!)

Reply:
@louh here's an example case; 
I own this street, but im not signed in. When I go to 'add location', I don't see any buttons to modify.
![notsignedin](https://user-images.githubusercontent.com/10539855/87457679-62381980-c5ce-11ea-9af8-21b755e0476c.gif)